community/primitive-collections/src/main/java/org/neo4j/collection/primitive/PrimitiveIntCollections.java

```diff
@@ -848,4 +840,33 @@ public static boolean contains( int[] values, int candidate )
             throw new IllegalStateException( "Encountered an already added item:" + item +
                     " when adding items uniquely to a collection:" + collection );
         }
-    }}
+    }
+
+    /**
+     * Deduplicates values in the {@code values} array.
+     *
+     * @param values sorted array of int values.
+     * @return the provided array if no duplicates were found, otherwise a new shorter array w/o duplicates.
```

Comment:
If you add a check like this, you can extract the actual deduplication to a helper method, to give the JIT more options for inlining and optimisation.